CMakeLists.txt

```diff
@@ -20,37 +20,37 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(VERSION 2.0.10)
-include(CheckCXXCompilerFlag)
-if (${CMAKE_CXX_COMPILER_ID} MATCHES "GNU")
-  set(CXX_FLAGS_TO_CHECK "-Wall -Wextra -fvisibility-inlines-hidden -fmessage-length=0 -ftree-vectorize -fstack-protector-strong -O2 -pipe -fext-numeric-literals")
-  set(CXX_FLAGS_DEBUG_TO_CHECK "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=address ${CMAKE_CXX_FLAGS_TO_CHECK}")
-  set(CMAKE_EXE_LINKER_FLAGS_DEBUG "-fsanitize=address")
-else()
-  set(CXX_FLAGS_TO_CHECK "-Wall -Wextra")
-  set(CXX_FLAGS_DEBUG_TO_CHECK "${CMAKE_CXX_FLAGS_DEBUG} -fsanitize=address ${CMAKE_CXX_FLAGS_TO_CHECK}")
-endif()
-set(CXX_FLAGS_PASSED )
-set(CXX_FLAGS_DEBUG_PASSED )
-foreach(fl ${CXX_FLAGS_TO_CHECK})
-  CHECK_CXX_COMPILER_FLAG(${fl} COMPILER_SUPPORTS_${fl})
-  if(COMPILER_SUPPORTS_${fl})
-    set(CXX_FLAGS_PASSED "${CXX_FLAGS_PASSED} ${fl}" )
+
+# Set default CXXFLAGS but allow for environment override
+# c.f. https://cmake.org/cmake/help/v3.31/envvar/CXXFLAGS.html
```

Comment:
First determine the values of `CMAKE_CXX_FLAGS`: Either taken from the environment `CXXFLAGS` like normal, or if they are absent, set defaults manually.